File: elixir/space-age/lib/space_age.ex
defmodule SpaceAge do
  @type planet ::
          :mercury
          | :venus
          | :earth
          | :mars
          | :jupiter
          | :saturn
          | :uranus
          | :neptune

  @seconds_per_earth_year 31_557_600

  @orbital_period %{
    mercury:   0.2408467,
    venus:     0.61519726,
    earth:     1.0,
    mars:      1.8808158,
    jupiter:  11.862615,
    saturn:   29.447498,
    uranus:   84.016846,
    neptune: 164.79132,
  }

  @doc """
  Return the number of years a person that has lived for 'seconds' seconds is
  aged on 'planet', or an error if 'planet' is not a planet.
  """
  @spec age_on(planet, pos_integer) :: {:ok, float} | {:error, String.t()}
  def age_on(planet, seconds) do
    case @orbital_period[planet] do
      nil -> {:error, "not a planet"}
      p   -> {:ok, seconds / @seconds_per_earth_year / p}
    end
  end
end

File: php/space-age/SpaceAge.php
<?php

declare(strict_types=1);

class SpaceAge
{
    private const RELATIVE_ORBIT = [
        'Mercury' =>   0.2408467,
        'Venus'   =>   0.61519726,
        'Earth'   =>   1.0,
        'Mars'    =>   1.8808158,
        'Jupiter' =>  11.862615,
        'Saturn'  =>  29.447498,
        'Uranus'  =>  84.016846,
        'Neptune' => 164.79132,
    ];
    private const SECONDS_PER_EARTH_YEAR = 31557600;
    private int $_seconds;

    public function __construct(int $seconds)
    {
        $this->_seconds = $seconds;
    }

    public function seconds(): int
    {
        return $this->_seconds;
    }

    public function __call(string $planet, array $args): float
    {
        $Planet = ucfirst($planet);
        if (!array_key_exists($Planet, self::RELATIVE_ORBIT)) {
            trigger_error("Error: $Planet is not a planet");
        }
        return $this->_seconds
                / self::RELATIVE_ORBIT[$Planet]
                / self::SECONDS_PER_EARTH_YEAR;
    }
}
